packages/twenty-front/src/modules/settings/security/components/SettingsSSOIdentitiesProvidersForm.tsx

```diff
@@ -0,0 +1,121 @@
+import { Controller, useFormContext } from 'react-hook-form';
+import { Section } from '@/ui/layout/section/components/Section';
+import { H2Title, IconKey } from 'twenty-ui';
+import { TextInput } from '@/ui/input/components/TextInput';
+import { SettingsAccountsRadioSettingsCard } from '@/settings/accounts/components/SettingsAccountsRadioSettingsCard';
+import { SettingsSSOOIDCForm } from '@/settings/security/components/SettingsSSOOIDCForm';
+import { SettingsSSOSAMLForm } from '@/settings/security/components/SettingsSSOSAMLForm';
+import { SettingsPageContainer } from '@/settings/components/SettingsPageContainer';
+import styled from '@emotion/styled';
+import { SettingSecurityNewSSOIdentityFormValues } from '@/settings/security/types/SSOIdentityProvider';
+import { IdpType } from '~/generated/graphql';
+import { ReactElement } from 'react';
+
+const StyledInputsContainer = styled.div`
+  display: grid;
+  gap: ${({ theme }) => theme.spacing(2, 4)};
+  grid-template-columns: 1fr 1fr;
+  grid-template-areas:
+    'input-1 input-1'
+    'input-2 input-3'
+    'input-4 input-5';
+
+  & :first-of-type {
+    grid-area: input-1;
+  }
+`;
+
+export const SettingsSSOIdentitiesProvidersForm = () => {
+  const { control, getValues } =
+    useFormContext<SettingSecurityNewSSOIdentityFormValues>();
+
+  const IdpMap: Record<
+    IdpType,
+    {
+      form: ReactElement;
+      card: {
+        cardMedia: ReactElement;
+        title: string;
+        value: string;
+        description: string;
+      };
+    }
+  > = {
+    OIDC: {
+      card: {
+        cardMedia: <IconKey />,
+        title: 'OIDC',
+        value: 'OIDC',
+        description: '',
+      },
+      form: <SettingsSSOOIDCForm />,
+    },
+    SAML: {
+      card: {
+        cardMedia: <IconKey />,
+        title: 'SAML',
+        value: 'SAML',
+        description: '',
+      },
+      form: <SettingsSSOSAMLForm />,
+    },
+  };
+
+  const getFormByType = (type: Uppercase<IdpType> | undefined) => {
+    switch (type) {
+      case IdpType.Oidc:
+        return IdpMap.OIDC.form;
+      case IdpType.Saml:
+        return IdpMap.SAML.form;
+      default:
+        return null;
+    }
+  };
+
+  return (
+    <SettingsPageContainer>
+      <Section>
+        <H2Title title="Name" description="The name of your connection" />
+        <StyledInputsContainer>
+          <Controller
```

Comment:
you could not have guessed but we actually don't want to use react hook form anymore. Let's keep it as it is, no need to rebuild your form

Reply:
Where can I find the best practice for managing form?